ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionSynchronizationInterceptor.java

```diff
@@ -90,6 +135,11 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
                 ROOT_LOGGER.trace("Acquired lock: " + lock + " for stateful component instance: " + instance + " during invocation: " + context);
             }
 
+            for (;;) {
```

Comment:
This spin loop is indicitive of a fundamental logic problem.  Maybe there should be a design discussion on the wildfly-dev list?